velocity/src/main/java/me/leoko/advancedban/velocity/VelocityMethods.java

```diff
@@ -0,0 +1,368 @@
+        layouts = LightningBuilder.fromFile(layoutFile).addInputStreamFromResource(layoutFile.getName()).setConfigSettings(ConfigSettings.PRESERVE_COMMENTS).createYaml();
+        if (mysqlFile.exists()) {
+            mysql = LightningBuilder.fromFile(mysqlFile).setConfigSettings(ConfigSettings.PRESERVE_COMMENTS).createYaml();
+        } else mysql = config;
+    }
+
+    @Override
+    public String getFromUrlJson(String url, String key) {
+        try {
+            HttpURLConnection request = (HttpURLConnection) new URL(url).openConnection();
+            request.connect();
+
+            JsonParser jp = new JsonParser();
+            JsonObject json = (JsonObject) jp.parse(new InputStreamReader(request.getInputStream()));
+
+            String[] keys = key.split("\\|");
+            for (int i = 0; i < keys.length - 1; i++) {
+                json = json.getAsJsonObject(keys[i]);
+            }
+
+            return json.get(keys[keys.length - 1]).toString().replaceAll("\"", "");
+
+        } catch (Exception exc) {
+            return null;
+        }
+    }
+
+    @Override
+    public String getVersion() {
+        String version = "Unknown";
+        try (InputStream stream = getClass().getClassLoader().getResourceAsStream("velocity-plugin.json"); InputStreamReader read = new InputStreamReader(stream)) {
+            JsonObject obj = new JsonParser().parse(read).getAsJsonObject();
+            version = obj.get("version").getAsString();
+        } catch (IOException ignored) { }
+        return version;
+    }
+
+    @Override
+    public String[] getKeys(Object file, String path) {
+        return ((Yaml) file).getSection(path).singleLayerKeySet().toArray(new String[0]);
+    }
+
+    @Override
+    public Yaml getConfig() {
+        return config;
+    }
+
+    @Override
+    public Yaml getMessages() {
+        return messages;
+    }
+
+    @Override
+    public Yaml getLayouts() {
+        return layouts;
+    }
+
+    @Override
+    public void setupMetrics() {
+        // Velocity does not support bStats
+    }
+
+    @Override
+    public boolean isBungee() {
+        return true;
+    }
+
+    @Override
+    public String clearFormatting(String text) {
+        return text.replace("§", "&");
+    }
+
+    @Override
+    public VelocityMain getPlugin() {
+        return VelocityMain.get();
+    }
+
+    @Override
+    public File getDataFolder() {
+        return getPlugin().folder;
+    }
+
+    @Override
+    public void setCommandExecutor(String cmd, TabCompleter tabCompleter) {
+        getPlugin().server.getCommandManager().register(new CommandReceiverVelocity(cmd), cmd);
+    }
+
+    @Override
+    public void sendMessage(Object player, String msg) {
+        if (player instanceof Player) {
+            ((Player) player).sendMessage(TextComponent.of(msg));
```

Comment:
Likewise with #hasPermission, this is an unnecessary cast. Also, are you sure `TextComponent.of` is the correct method? I thought the LegacyComponentSerializer was supposed to be used, although it's possible TextComponent.of will work for now.